File: JavaScript/8js 1/aliasGen.js
/*
|--------------------------------------------------------------------------
| Crash Override
|--------------------------------------------------------------------------
|
*/

function aliasGen(x, y) {
   return /^[a-z]/i.test(x) && /^[a-z]/i.test(y)
      ? `${firstName[x[0].toUpperCase()]} ${surname[y[0].toUpperCase()]}`
      : "Your name must start with a letter from A - Z.";
}


function aliasGen(f, l) {
   const n1 = firstName[f[0].toUpperCase()];
   const n2 = surname[l[0].toUpperCase()];
   return n1 && n2 ? `${n1} ${n2}` : "Your name must start with a letter from A - Z.";
}


function aliasGen(name, sname) {
   if (+name[0] || +sname[0]) {
      return "Your name must start with a letter from A - Z.";
   }
   return firstName[name[0].toUpperCase()] + " " + surname[sname[0].toUpperCase()];
}


console.log(aliasGen("Mike", "Millington"), "Malware Mike");
console.log(aliasGen("Fahima", "Tash"), "Function T-Rex");
console.log(aliasGen("Daisy", "Petrovic"), "Data Payload");
console.log(aliasGen("7393424", "Anumbha"), "Your name must start with a letter from A - Z.");
console.log(aliasGen("Anuddanumbha", "23200"), "Your name must start with a letter from A - Z.");

File: Python/8py 1/alias_gen.py
"""
|--------------------------------------------------------------------------
| Crash Override
|--------------------------------------------------------------------------
|
"""

def alias_gen(f_name, l_name):
   try:
      return f"{FIRST_NAME[f_name[0].upper()]} {SURNAME[l_name[0].upper()]}"
   except:
      return "Your name must start with a letter from A - Z."
    

from re import findall
def alias_gen(f, l):
   if findall(r'\b\d', f+' '+l): 
      return 'Your name must start with a letter from A - Z.'
   else: 
      return f'{FIRST_NAME[f[0].upper()]} {SURNAME[l[0].upper()]}'
   

def alias_gen(f_name, l_name):
   if f_name[0].isalpha() and l_name[0].isalpha():
      return "{} {}".format(FIRST_NAME[f_name[0].upper()], SURNAME[l_name[0].upper()])
   else:
      return "Your name must start with a letter from A - Z."
   

print(alias_gen("Mike", "Millington"), "Malware Mike")
print(alias_gen("Fahima", "Tash"), "Function T-Rex")
print(alias_gen("Daisy", "Petrovic"), "Data Payload")
print(alias_gen("7393424", "Anumbha"), "Your name must start with a letter from A - Z.")
print(alias_gen("Anuddanumbha", "23200"), "Your name must start with a letter from A - Z.")